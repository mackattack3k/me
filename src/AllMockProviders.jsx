import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import defaultTheme from './features/theme/defaultTheme';
import GithubMockProvider from './testproviders/GithubMockProvider';

const AllMockProviders = ({ children }) => (
  <ThemeProvider theme={defaultTheme.dark}>
    <GithubMockProvider>{children}</GithubMockProvider>
  </ThemeProvider>
);

AllMockProviders.propTypes = {
  children: PropTypes.node.isRequired
};

export default AllMockProviders;
