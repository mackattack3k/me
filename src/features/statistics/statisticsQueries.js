import { gql } from 'apollo-boost';

export const GITHUB_CONTRIBUTIONS = gql`
  {
    user(login: "mackattack3k") {
      login
      contributionsCollection {
        totalCommitContributions
        commitContributionsByRepository {
          repository {
            name
          }
        }
      }
      starredRepositories(
        first: 1
        orderBy: { direction: DESC, field: STARRED_AT }
      ) {
        nodes {
          description
          name
          owner {
            login
            url
          }
          url
        }
      }
    }
    search(type: ISSUE, first: 10, query: "author:mackattack3k") {
      issueCount
    }
  }
`;
