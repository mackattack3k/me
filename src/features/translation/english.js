export default {
  routes: {
    home: 'home',
    contact: 'contact',
    statistics: 'cool stuff',
    settings: 'settings'
  },
  home: {
    title1: 'Marcus is the best',
    section1:
      'Marcus is amazing and knows how to code good very good. Like very good.',
    title2: 'git gud',
    section2: 'When you know git as well as Marcus. Nothing is impossible.',
    title3: 'I want to play a game',
    section3: 'Do you want to see some cool data? Click here'
  },
  theme: {
    changeThemeTitle: 'Make it easier on your eyes',
    darkTheme: 'Dark theme',
    changeLanguageTitle: 'Snackar du svenska?',
    changeLanguage: 'Ja!'
  },
  statistics: {
    github:
      'Wondering what Marcus has been up to on github? Check out this compact view!'
  },
  git: {
    total_commits: 'Commits since last year',
    issue_count: 'Issues closed',
    latest_star: 'Is the latest starred repository'
  }
};
