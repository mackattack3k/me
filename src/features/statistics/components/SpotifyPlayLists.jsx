import { useQuery } from '@apollo/react-hooks';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import ErrorPage from '../../error/components/ErrorPage';
import LoadingSpinner from '../../loading/components/LoadingSpinner';
import H3 from '../../text/components/H3';
import { SPOTIFY_USER_PLAYLISTS } from '../statisticsQueries';
import StatisticsCard from './StatisticsCard';

const PlayLists = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

const SpotifyPlayLists = () => {
  const { t } = useTranslation();
  const { loading, error, data } = useQuery(SPOTIFY_USER_PLAYLISTS);
  if (error) {
    return <ErrorPage />;
  }
  const { playlist: playlists = {} } = data || {};
  const { items = [] } = playlists;
  return (
    <div>
      <H3>{t('spotify_playlist.title')}</H3>
      {loading ? (
        <LoadingSpinner />
      ) : (
        <PlayLists>
          {items.map(playlist => (
            <StatisticsCard key={playlist.id}>{playlist.name}</StatisticsCard>
          ))}
        </PlayLists>
      )}
    </div>
  );
};

SpotifyPlayLists.propTypes = {};

export default SpotifyPlayLists;
