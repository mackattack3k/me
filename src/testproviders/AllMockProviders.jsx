import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../features/theme/defaultTheme';
import { ToastProvider } from '../features/toast/components/ToastProvider';
import GithubMockProvider from './GithubMockProvider';

const AllMockProviders = ({ children }) => (
  <ThemeProvider theme={defaultTheme.dark}>
    <ToastProvider>
      <GithubMockProvider>{children}</GithubMockProvider>
    </ToastProvider>
  </ThemeProvider>
);

AllMockProviders.propTypes = {
  children: PropTypes.node.isRequired
};

export default AllMockProviders;
