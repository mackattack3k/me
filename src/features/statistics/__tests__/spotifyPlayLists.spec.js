import { MockedProvider } from '@apollo/client/testing';
import { act, render } from '@testing-library/react';
import React from 'react';
import wait from 'waait';
import AllMockProviders from '../../../testproviders/AllMockProviders';
import english from '../../translation/english';
import SpotifyPlayLists from '../components/SpotifyPlayLists';
import { SPOTIFY_USER_PLAYLISTS } from '../statisticsQueries';

const mountMockedProvider = async () => {
  const component = render(
    <AllMockProviders>
      <SpotifyPlayLists />
    </AllMockProviders>
  );

  await act(() => wait(0));
  return component;
};
const errorMockProvider = async () => {
  const spotifyErrorMock = {
    request: {
      query: SPOTIFY_USER_PLAYLISTS
    },
    error: new Error('aw shucks')
  };
  const component = render(
    <AllMockProviders>
      <MockedProvider mocks={[spotifyErrorMock]}>
        <SpotifyPlayLists />
      </MockedProvider>
    </AllMockProviders>
  );
  await act(() => wait(0));
  return component;
};

describe('spotify users playlists', () => {
  it('should render with snapshot', async () => {
    const spotifyPlaylists = await mountMockedProvider();
    expect(spotifyPlaylists).toMatchSnapshot();
  });
  it('should show error on data failure', async () => {
    const { getByText } = await errorMockProvider();
    const errorMessage = english.error.unknown;
    const error = getByText(new RegExp(errorMessage, 'g'));
    expect(error).toBeDefined();
  });
});
