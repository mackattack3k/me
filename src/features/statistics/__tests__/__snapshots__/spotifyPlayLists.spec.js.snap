// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spotify users playlists should render with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  position: absolute;
  bottom: 0;
  left: 0;
}

.c0 {
  font-family: 'Proza Libre',sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<body>
    <div>
      <div>
        <h3
          class="c0"
        >
          Playlists
        </h3>
        <div
          class="c1"
        />
      </div>
      <div
        class="c2"
      />
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        class="sc-AxhCb bIlqnU"
      >
        Playlists
      </h3>
      <div
        class="sc-AxmLO hYkXfO"
      />
    </div>
    <div
      class="sc-AxirZ ejZzxV"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
